packages/ember-views/lib/views/view.js

```diff
@@ -1791,7 +1791,7 @@ var View = CoreView.extend({
     Ember.assert("Only arrays are allowed for 'classNameBindings'", typeOf(this.classNameBindings) === 'array');
     this.classNameBindings = emberA(this.classNameBindings.slice());
 
-    Ember.assert("Only arrays are allowed for 'classNames'", typeOf(this.classNames) === 'array');
+    Ember.assert("Only arrays are allowed for 'classNames'. To bind class names to computed properties, use 'classNameBindings' instead.", typeOf(this.classNames) === 'array');
```

Comment:
@rwjblue I updated the wording (yeah, I think that makes the same point); the problem with a space-separated string, though, is that this assertion actually doesn't throw an error in that case, since `classNames` is a concatenated property. If you use a space-separated string from `classNames`, it will simply add that entire string to the `classNames` array, and this assertion will still pass. I tested this to verify it. (This is probably something that could be fixed/tested separately.)